docs/server-side-search.rst

```diff
@@ -64,3 +64,107 @@ and then click on :guilabel:`Search Analytics`.
    Search analytics demo
 
 .. _Elasticsearch: https://www.elastic.co/products/elasticsearch
+
+API
+---
+
+Search is exposed through our API that's proxied from the domain where your docs are being served.
+This is ``https://docs.readthedocs.io/_/api/v2/docsearch`` for the ``docs`` project, for example.
+
+.. http:get:: /_/api/v2/docsearch/
+
+   Return a list of search results for a project,
+   including results from its :doc:`/subprojects`.
+   Results are divided into sections with highlights of the matching term.
+
+   **Example request**:
+
+   .. tabs::
+
+      .. code-tab:: bash
+
+         $ curl "https://docs.readthedocs.io/_/api/v2/docsearch/?project=docs&version=latest&q=server%20side%20search"
+
+      .. code-tab:: python
+
+         import requests
+         URL = 'https://docs.readthedocs.io/_/api/v2/docsearch/'
+         params = {
+            'q': 'server side search',
+            'project': 'docs',
+            'version': 'latest',
+         }
+         response = requests.get(URL, params=params)
+         print(response.json())
+
+   :query q: Search query
+   :query project: Project slug
+   :query version: Version slug
+
+   **Example response**:
+
+   .. sourcecode:: json
+
+      {
```

Comment:
We should probably be defining all the fields here that are coming back with the HTTP module, like this: https://github.com/readthedocs/ethical-ad-server/blob/baf4260c1c5fd063e7281b1ba5c9c0970b9b8908/adserver/api/views.py#L70-L80